skimage/segmentation/_slic.pyx

```diff
@@ -137,25 +152,31 @@ def _slic_cython(np_floats[:, :, :, ::1] image_zyx,
                         dy = sy * (cy - y)
                         dy *= dy
                         for x in range(x_min, x_max):
+
+                            if use_mask and not mask[z, y, x]:
+                                continue
+
                             dx = sx * (cx - x)
                             dx *= dx
                             dist_center = (dz + dy + dx) * spatial_weight
                             dist_color = 0
                             for c in range(3, n_features):
                                 t = image_zyx[z, y, x, c - 3] - segments[k, c]
                                 dist_color += t * t
+
                             if slic_zero:
-                                dist_center += dist_color / max_dist_color[k]
-                            else:
+                                dist_color /= max_dist_color[k]
+
```

Comment:
It looks like this color distance computation can be moved within the `if not ignore_color` block below